posthog/hogql_queries/web_analytics/external_clicks.py

```diff
@@ -0,0 +1,119 @@
+from posthog.hogql import ast
+from posthog.hogql.constants import LimitContext
+from posthog.hogql.parser import parse_select
+from posthog.hogql.property import (
+    property_to_expr,
+)
+from posthog.hogql_queries.insights.paginators import HogQLHasMorePaginator
+from posthog.hogql_queries.web_analytics.web_analytics_query_runner import (
+    WebAnalyticsQueryRunner,
+    map_columns,
+)
+from posthog.schema import (
+    CachedWebStatsTableQueryResponse,
+    WebStatsTableQueryResponse,
+    WebExternalClicksTableQuery,
+    WebExternalClicksTableQueryResponse,
+)
+
+
+class WebExternalClicksTableQueryRunner(WebAnalyticsQueryRunner):
+    query: WebExternalClicksTableQuery
+    response: WebExternalClicksTableQueryResponse
+    cached_response: CachedWebStatsTableQueryResponse
+    paginator: HogQLHasMorePaginator
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.paginator = HogQLHasMorePaginator.from_limit_context(
+            limit_context=LimitContext.QUERY, limit=self.query.limit if self.query.limit else None
+        )
+
+    def to_query(self) -> ast.SelectQuery:
+        if self.query.stripQueryParams:
+            url_expr: ast.Expr = ast.Call(
+                name="cutQueryStringAndFragment",
+                args=[ast.Field(chain=["properties", "$external_click_url"])],
+            )
+        else:
+            url_expr = ast.Field(chain=["properties", "$external_click_url"])
+
+        with self.timings.measure("stats_table_query"):
+            query = parse_select(
```

Comment:
Do we know the performance impact of parsing? Feels like this could be rewritten as an AST

Reply:
On my laptop:

```
{
  "timings": [
        {
            "k": "./stats_table_query/parse_select_cpp/replace_placeholders",
            "t": 0.00017500000103609636
        },
        {
            "k": "./stats_table_query/parse_select_cpp",
            "t": 0.011916250001377193
        },
        {
            "k": "./stats_table_query",
            "t": 0.012091958000382874
        },
        {
            "k": "./query",
            "t": 0.000013500000932253897
        },
        {
            "k": "./replace_placeholders",
            "t": 0.00009395899905939586
        },
        {
            "k": "./max_limit",
            "t": 9.291998139815405e-6
        },
        {
            "k": "./hogql/prepare_ast/clone",
            "t": 0.0000960830002441071
        },
        {
            "k": "./hogql/prepare_ast/create_hogql_database",
            "t": 0.09500029099945095
        },
        {
            "k": "./hogql/prepare_ast/resolve_types",
            "t": 0.0004964589970768429
        },
        {
            "k": "./hogql/prepare_ast",
            "t": 0.095637333997729
        },
        {
            "k": "./hogql/print_ast/printer",
            "t": 0.00025674999778857455
        },
        {
            "k": "./hogql/print_ast",
            "t": 0.0002847079995262902
        },
        {
            "k": "./hogql",
            "t": 0.09593283400317887
        },
        {
            "k": "./print_ast/create_hogql_database",
            "t": 0.08269116600058624
        },
        {
            "k": "./print_ast/resolve_types",
            "t": 0.0005057919988757931
        },
        {
            "k": "./print_ast/resolve_property_types",
            "t": 0.0020862500023213215
        },
        {
            "k": "./print_ast/resolve_lazy_tables",
            "t": 0.0010285409989592154
        },
        {
            "k": "./print_ast/swap_properties",
            "t": 0.00021833299979334697
        },
        {
            "k": "./print_ast/printer",
            "t": 0.0005697500018868595
        },
        {
            "k": "./print_ast",
            "t": 0.08716541700050584
        },
        {
            "k": "./clickhouse_execute",
            "t": 0.05190262500036624
        },
        {
            "k": ".",
            "t": 0.2540554580009484
        }
    ],
}
```

If we ignore the clickhouse_execute part, this part is about 6% of the request (0.012091958000382874 / (0.2540554580009484 - 0.05190262500036624) = 0.05981592155). Not nothing, but IMO worth it to make the query more readable.